### ISSUE TITLE
phoenicis.sh must be different for Java 8 and 10/11

### ISSUE THREAD

**Issue by plata** [PhoenicisOrg/phoenicis#1408]
For Java 10/11, run with:
```
java -cp 'phoenicis-dist/target/lib/*':phoenicis-dist/target/phoenicis-dist-5.0-SNAPSHOT.jar -p /<path>/javafx-sdk-11/lib --add-modules javafx.controls org.phoenicis.javafx.JavaFXApplication
```

How can we handle this?

**Comment by madoar** [comment]
I'm not sure if we talked about this before, but just in case:
Why don't we provide two jar archives, containing our applications with all necessary dependencies.
The first jar archive would contain the javafx application, while the second one contains the cli application.
If we have this structure, we should be able to start the application with `java -jar application.jar`, independent of the available Java version.

**Comment by madoar** [comment]
About the modules:
I think we should try to add a `module-info.java` file to our subprojects and see if this breaks our compatibility with Java 8.

**Comment by plata** [comment]
I'm not sure if it makes sense to implement modules as long as we support Java 8. Apart from this, I don't really see how this would solve the problem.

**Comment by plata** [comment]
I'm not really familiar with the module system in Java. Do you want to try it?

In any case, I've figured out the required exports:
```
--add-exports javafx.graphics/com.sun.javafx.sg.prism=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.scene=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.util=ALL-UNNAMED
--add-exports javafx.base/com.sun.javafx.logging=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.prism=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.glass.ui=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.geom.transform=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.glass.utils=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.font=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.application=ALL-UNNAMED
--add-exports javafx.controls/com.sun.javafx.scene.control=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.scene.input=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.geom=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.prism.paint=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.scenario.effect=ALL-UNNAMED
--add-exports javafx.graphics/com.sun.javafx.text=ALL-UNNAMED
```

**Comment by madoar** [comment]
I can take a look at this, but it's hard to test it without Java 11 installed, especially if it needs to be manually tested.
For the beginning I've found this [tutorial](https://dzone.com/articles/javafx-on-jdk-11), where the use of JavaFX from JDK11 with the module system is explained.

**Comment by plata** [comment]
I've recently upgraded to Mint 19 (Java 10 is default).

**Comment by plata** [comment]
Regarding the module-info: https://maven.apache.org/plugins/maven-compiler-plugin/examples/module-info.html

**Comment by ImperatorS79** [comment]
What should be <path> in the command ?

**Comment by plata** [comment]
It's the path to where you've downloaded OpenJFX. I thought this shouldn't be required but I couldn't make it run with the Maven dependencies only.

**Comment by ImperatorS79** [comment]
So I have to downoad openjfx myself ?

**Comment by ImperatorS79** [comment]
Just for your info, I managed to launch phoenicis on Ubuntu 18.10 (which has openjdk and openjfx 11) with:
```sh
#!/usr/bin/env bash
POL_HOME="$(dirname "$0")"
cd "$POL_HOME"
POL_HOME="$PWD"
CLASSPATH=${CLASSPATH}:$POL_HOME/lib/*

java -cp "$CLASSPATH":$POL_HOME/../phoenicis-dist-5.0-SNAPSHOT.jar -p /usr/share/openjfx/lib/ --add-modules javafx.controls org.phoenicis.javafx.JavaFXApplication
```

**Comment by plata** [comment]
Thanks for the info. The question is, how can we handle the OpenJFX path if it must be installed manually. I still don't understand why maven is not enough.

**Comment by ImperatorS79** [comment]
Just for your info 2: I have just removed every javafx jar in the lib directory and my phoenicis.sh worked fine.

**Comment by plata** [comment]
Does it also work the other way around (i.e. remove your local JavaFX installation and leave the jars in the lib folder)?

**Comment by ImperatorS79** [comment]
It works too if I do:
```bash
#!/usr/bin/env bash
POL_HOME="$(dirname "$0")"
cd "$POL_HOME"
POL_HOME="$PWD"
CLASSPATH=${CLASSPATH}:$POL_HOME/lib/*

java -cp "$CLASSPATH":$POL_HOME/../phoenicis-dist-5.0-SNAPSHOT.jar -p $POL_HOME/javafx/ --add-modules javafx.controls org.phoenicis.javafx.JavaFXApplication
```
where `$POL_HOME/javafx` is where i have placed the javafx libs compiled by maven

**Comment by plata** [comment]
Shouldn't the JavaFX libs be placed in the lib directory? Is it not sufficient if they are in the classpath?

**Comment by ImperatorS79** [comment]
If I do:
```bash
#!/usr/bin/env bash
POL_HOME="$(dirname "$0")"
cd "$POL_HOME"
POL_HOME="$PWD"
CLASSPATH=${CLASSPATH}:$POL_HOME/lib/*

java -cp "$CLASSPATH":$POL_HOME/../phoenicis-dist-5.0-SNAPSHOT.jar -p $POL_HOME/lib/ --add-modules javafx.controls org.phoenicis.javafx.JavaFXApplication
```

I have:
```
Error occurred during initialization of boot layer
java.lang.module.ResolutionException: Modules xml.apis and xmlParserAPIs export package org.xml.sax.ext to module jmimemagic
```

**Comment by plata** [comment]
That can be fixed (check the flatpak pr). Not sure what's the better solution though.

**Comment by madoar** [comment]
I think we'll need to try a minimal example from https://openjfx.io/openjfx-docs/ and see if it works at all for us.
Then we'll need to see which project changes are required.

**Comment by plata** [comment]
We probably need to scripts and then we can have two profiles (JDK 8 and JDK >= 10) to copy and rename the according script in phoenicis-dist.

**Comment by intenseboy** [comment]
For IntelliJIDEA : 
Go to Run/Debug Configurations and add this to[VM options:]
--module-path /YourPathToJavaFxLibs/javafx-sdk-11.0.1/lib --add-modules=javafx.controls,javafx.web,javafx.fxml

**Comment by qparis** [comment]
> For IntelliJIDEA :
> Go to Run/Debug Configurations and add this to[VM options:]
> --module-path /YourPathToJavaFxLibs/javafx-sdk-11.0.1/lib --add-modules=javafx.controls,javafx.web,javafx.fxml

javafx.controls,javafx.web and javafx.xml are not enough, ALL_MODULE_PATH is better: 
https://phoenicisorg.github.io/phoenicis/Developers/intellij-idea/

**Comment by Kreyren** [comment]
tldr? Seems like solution for an if statement that grebs java version from JAVA_HOME..
